components/authz-service/server/v2/project.go

```diff
@@ -568,6 +541,65 @@ func (s *ProjectState) DeleteRule(ctx context.Context, req *api.DeleteRuleReq) (
 	}
 }
 
+const ruleSetEditsPendingStatus = "edits-pending"
+
+func (s *ProjectState) validateProjectDelete(ctx context.Context, projectID string) error {
+	projectResponse, err := s.GetProject(ctx, &api.GetProjectReq{Id: projectID})
+	if err != nil {
+		return err
+	}
+
+	if projectResponse.Project.Status == ruleSetEditsPendingStatus {
```

Comment:
The project can not be deleted if it has 'edits-pending'.

Reply:
like @blakestier mentioned at standup, it should be ok to delete a project if there's only staged rules, because they're not in effect and they get blown away on project delete. however, since we're worried about race conditions with the project update, it's probably best to be more restrictive, as you've done here.